src/Kiota.Builder/Refiners/CommonLanguageRefiner.cs

```diff
@@ -331,26 +343,26 @@ private static CodeTypeBase ConvertUnionTypeToWrapper(CodeClass codeClass, CodeU
         if(codeUnionType.Types.All(x => x.TypeDefinition is CodeClass targetClass && targetClass.IsOfKind(CodeClassKind.Model) ||
                                 x.TypeDefinition is CodeEnum)) {
             KiotaBuilder.AddSerializationMembers(newClass, true, usesBackingStore);
-            newClass.Kind = CodeClassKind.Model;
+            if (newClass != null) newClass.Kind = CodeClassKind.Model;
         }
         // Add the discrimnator function to the wrapper as it will be referenced. 
-        var factoryMethod = newClass.AddMethod(new CodeMethod
+        var factoryMethod = newClass?.AddMethod(new CodeMethod
         {
             Name = "CreateFromDiscriminatorValue",
             ReturnType = new CodeType { TypeDefinition = newClass, Name = newClass.Name, IsNullable = false },
             Kind = CodeMethodKind.Factory,
             IsStatic = true,
             IsAsync = false,
         }).First();
-        factoryMethod.AddParameter(new CodeParameter
+        factoryMethod?.AddParameter(new CodeParameter
```

Comment:
assuming the newClass is not null, the factory method can't, so no need for the null propagation operator here